### PR TITLE
Initial version - Added all relevant current sources to Nagios4Nonstop open-source project

### PR DESCRIPTION
**Pull request by HPE-RGHEGDE**
- License headers, README documentation are updated

### CODE REVIEW

File: bin/get_ns_tmf_tps.pl [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+#   if ( $buf =~ /(.*?)([0-9]+\.[0-9]+)\s*TPS/s ){
+   if ( $buf =~ /(.*?)(\d+)\s*TPS/s ){
+	   my $t=$1; #This is TPS
+	   print "t value = ".$t."\n";
+	   $t = sprintf("%f",$t) ; # Don't round off.
+	   $perf_data.="TPS= ".$t;
+	   $main_data.="Transaction Rate = ".$t." TPS";
+   }
+
+=head not required
```
Comment (siddharthagudgunti) [inline bin/get_ns_tmf_tps.pl]: Commented out lines can be removed. There are other files also, where such lines present.

File: bin/get_ns_zzsa_count.pl [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+# ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# 
+# The views and conclusions contained in the software and documentation are those of the
+# authors and should not be interpreted as representing official policies, either expressed
+# or implied, of Hewlett Packard Enterprise Development LP.
+########################################################################################
+
+###############################################################################
```
Comment (siddharthagudgunti) [inline bin/get_ns_zzsa_count.pl:29]: I see in some files this Author info is present. If you want to keep it, then make it consistent across files, else just remove. I personally prefer removing Author field, but I like the Purpose section, it provides brief intent of script.

File: etc/ns_template [hunk 3]
```diff
@@ -0,0 +1,160 @@
+define host {
+    use 			    generic-host
+    host_name                       myhostname
+    alias                           NS1
+    address                         MYHOST.IN.RDLABS.HPECORP.NET
```
Comment (siddharthagudgunti) [inline etc/ns_template]: Please generalize it like <myhost.fqdn>

File: etc/nscommands [hunk 4]
```diff
@@ -0,0 +1,55 @@
+
```
Comment (siddharthagudgunti) [inline etc/nscommands:1]: This file need Copyright info, similarly there are some other files, where it is missing, please add accordingly.

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+#Create softlink to ns/bin folder ..
+ln -f -s $nonstop_plugin_location/bin  /usr/local/ns/bin
+
+#use different file name in ns_meascom.cgi -- TBD later.  $SYSTEM.TMP.DATA01 and $SYSTEM.TMP.F1 
+
+#copy RRD files from the sources
+cp -a $nonstop_plugin_location/rrd /opt/nagiosgraph/var/
+
+#Set time zone
+ln -f -s  /usr/share/zoneinfo/Asia/Kolkata /etc/localtime
```
Comment (siddharthagudgunti) [inline install.sh]: This should be made dynamic as per user timezone. Either take it as parameter input or Environment variable.

File: rrd/README.txt [hunk 6]
```diff
@@ -0,0 +1,9 @@
+This is an empty folder. This will be copied to Nagios Graph location.
+
+If you are migrating an instance of an existing Nagios, Populate this folder with contents from */rrd/ folder from existing Nagios instance.
+
+Please ensure that configdb also is copied to nonstop_extensions/etc/ path.
```
Comment (siddharthagudgunti) [inline rrd/README.txt:5]: Maybe this requires correct path updates here?

File: sbin/addhosts.cgi [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+our $buf="
+<form method='post' action='receivensdata.cgi'>
+<INPUT TYPE='hidden' NAME='nagFormId' VALUE='b308ca00'>
+<table CELLSPACING=0 CELLPADDING=5>
+<tr>
+</tr>
+<tr>
+<td>Host Name: </td>
+<td><b><INPUT TYPE='TEXT' NAME='host' VALUE='brandy'></b></td>
```
Comment (siddharthagudgunti) [inline sbin/addhosts.cgi]: We should generalize the hostname here (like myhost maybe?).

File: sbin/addhosts.cgi [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+</tr>
+<tr>
+<td>User Name: </td>
+<td><b><INPUT TYPE='TEXT' NAME='user' VALUE='SUPER.SUPER' READONLY DISABLED ></b></td>
```
Comment (siddharthagudgunti) [inline sbin/addhosts.cgi:64]: All instances of user, please take it from a configuration file (which has been properly secured).

File: sbin/addhosts.cgi [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+</tr>
+<tr>
+<td>Password:&nbsp;&nbsp;&nbsp;</td>
+<td><b><INPUT TYPE='TEXT' NAME='password' VALUE='password'></b></td>
```
Comment (siddharthagudgunti) [inline sbin/addhosts.cgi:68]: We should not have anyplace to have user credentials stored, it should be dynamically queried/authenticated from system service.

File: sbin/receivensdata.cgi [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+   print "Updated config files for new host ($host).. <br>";
+#use tempate and update hostnames and create a new file 
+   my $simple_hostname=$host;
+   $simple_hostname =~ s/\..*//;
+#Copying the template as required hostname's file
+   print "simple hostname=$simple_hostname<br>";
+   my $tout=`cp -f $configtemplate_file $configtemplate_dir/$simple_hostname.cfg 2>&1  `;
+   print "cp output |$tout|<br>";
+#updating the hostname details 
+   $tout=`sed -i 's/MYHOST.IN.RDLABS.HPECORP.NET/$host/' $configtemplate_dir/$simple_hostname.cfg 2>&1`;
```
Comment (siddharthagudgunti) [inline sbin/receivensdata.cgi]: fqdn should be generalized

**Review by siddharthagudgunti — CHANGES_REQUESTED** [review]
Looks good. The Generic comments are around commented blocks of code.
Specific comments are related to user authentication.

File: bin/get_ns_tmf_tps.pl (hunk 1 above)
Comment (HPE-RGHEGDE) [reply bin/get_ns_tmf_tps.pl]: Thanks. Fixed this generically. Will let you know once the changes are pushed in the branch

File: etc/ns_template (hunk 3 above)
Comment (HPE-RGHEGDE) [reply etc/ns_template]: Done - generically

File: sbin/addhosts.cgi (hunk 7 above)
Comment (HPE-RGHEGDE) [reply sbin/addhosts.cgi]: Done

File: sbin/receivensdata.cgi (hunk 10 above)
Comment (HPE-RGHEGDE) [reply sbin/receivensdata.cgi]: Done

File: sbin/addhosts.cgi (hunk 9 above)
Comment (HPE-RGHEGDE) [reply sbin/addhosts.cgi:68]: Currently, All information is retrieved from nonstop, based on the interactive login. And this needs username/password to be available in plain text. Supporting password-less login to fetch required information is on the cards - But that will be implemented later. For now, To let the user know about this, I have added required notes on the page where we collect this information from the user. So - This comment will be also be tracked as an open issue on this GITHUB project.

File: etc/nscommands (hunk 4 above)
Comment (HPE-RGHEGDE) [reply etc/nscommands:1]: This does not need a license, This is a Nagios way of defining commands. It is basically a configuration file.

File: install.sh (hunk 5 above)
Comment (HPE-RGHEGDE) [reply install.sh]: This was not required. It's best it is taken from the default as configured on the system. Hence, Removed it.

File: rrd/README.txt (hunk 6 above)
Comment (HPE-RGHEGDE) [reply rrd/README.txt:5]: No. This path is correct.  The path is relative to path where main Nagios is deployed.

**Review by siddharthagudgunti — APPROVED** [review]
I do not have any more comments.

**Pull request merged**